File: Retos/Reto #0 - EL FAMOSO FIZZ BUZZ [Fácil]/python/dPenedo.py
"""
 * Escribe un programa que muestre por consola (con un print) los
 * números de 1 a 100 (ambos incluidos y con un salto de línea entre
 * cada impresión), sustituyendo los siguientes:
 * - Múltiplos de 3 por la palabra "fizz".
 * - Múltiplos de 5 por la palabra "buzz".
 * - Múltiplos de 3 y de 5 a la vez por la palabra "fizzbuzz".
"""


def es_multiplo(numero, divisor):
    return numero % divisor == 0

for i in range(1, 101):
    multiplo_de_3 = es_multiplo(i, 3)
    multiplo_de_5 = es_multiplo(i, 5)
    if multiplo_de_3 and multiplo_de_5:
        print("fizzbuzz")
    elif multiplo_de_3:
        print("fizz")
    elif multiplo_de_5:
        print("buzz")
    print(i)

File: Retos/Reto #0 - EL FAMOSO FIZZ BUZZ [Fácil]/javascript/dPenedo.js
// /
//   Escribe un programa que muestre por consola (con un print) los
//   números de 1 a 100 (ambos incluidos y con un salto de línea entre
//   cada impresión), sustituyendo los siguientes:
//   - Múltiplos de 3 por la palabra "fizz".
//   - Múltiplos de 5 por la palabra "buzz".
//   - Múltiplos de 3 y de 5 a la vez por la palabra "fizzbuzz".
//  

function isMultiple(number, divider){
    return number % divider == 0
}


function printNumbers(number){
    const listNumber = number
    for (let index = 1; index <= listNumber; index++) {
        const multipleOf3 = isMultiple(index, 3) 
        const multipleOf5 = isMultiple(index, 5) 
        if (multipleOf3 && multipleOf5){
            console.log("fizzbuzz")
        }else if(multipleOf3){
            console.log("fizz")
        }else if(multipleOf5){
            console.log("buzz")
        }
        else{
            console.log(index)
        }
    }
}

printNumbers(100)